Sources/FoundationEssentials/CMakeLists.txt

```diff
@@ -50,8 +50,26 @@ add_subdirectory(TimeZone)
 add_subdirectory(URL)
 
 if(SwiftFoundation_MACRO)
-  target_compile_options(FoundationEssentials PRIVATE
-    "SHELL:-plugin-path ${SwiftFoundation_MACRO}")
+    message(STATUS "SwiftFoundation_MACRO provided, using macros in ${SwiftFoundation_MACRO}")
+    # A path to Foundation macros was provided, so we use that path
+    target_compile_options(FoundationEssentials PRIVATE
+        "SHELL:-plugin-path ${SwiftFoundation_MACRO}")
+else()
+    message(STATUS "SwiftFoundation_MACRO not provided, building Foundation macros locally for host")
+    # No path to Foundation macros was provided, so we must build it ourselves
+    set(FoundationMacros_BuildLocalExecutable YES)
+    FetchContent_Declare(FoundationMacros
```

Comment:
The macros should build through ExternalProject. If we are cross-compiling Foundation, these macros will get built for the machine that we are cross-compiling to instead of the machine running the build, and therefore will not be usable.